pycbc/events/stat.py

```diff
@@ -1174,26 +1182,41 @@ def coinc_multiifo(self, s, slide, step, to_shift,
         benchmark_logvol = s[0][1]['benchmark_logvol']
         network_logvol -= benchmark_logvol
 
+        # Use prior histogram to get log signal rate
         stat = {ifo: st for ifo, st in s}
         logr_s = self.logsignalrate_multiifo(stat,
                                              slide * step, to_shift)
 
-        loglr = logr_s + network_logvol - ln_noise_rate
+        # Calculate noise rate in each bin given uniform phase time diffs
+        # and SNR ratio.
+        # Calculate bin & histogram volumes
+        n_ifos = len(self.hist_ifos)
+        # Effective histogram volume, as not all bins will be occupied by
+        # noise coincs for n_ifos > 2
+        noise_twindow = coinc_rate.multiifo_noise_coincident_area(
+                            self.hist_ifos, kwargs['time_addition'])
+        hist_vol = noise_twindow * (2 * numpy.pi * (self.srbmax - self.srbmin)
+                                    * self.swidth) ** (n_ifos - 1)
+
+        logr_n = - numpy.log(hist_vol)
+
+        # Combine to get final statistic
+        loglr = logr_s - logr_n + network_logvol - ln_noise_rate
```

Comment:
this seems to have two noise rate terms, logr_n and ln_noise rate, I don't immediately see how they are supposed to fit together